reposerver/repository/repository.go

```diff
@@ -1024,6 +1045,11 @@ func GenerateManifests(ctx context.Context, appPath, repoRoot, revision string,
 
 	resourceTracking := argo.NewResourceTracking()
 
+	// Set path of the source in the environment variable if ref field is set
+	if q.ApplicationSource.Ref != "" {
+		os.Setenv(q.ApplicationSource.Ref, appPath)
+	}
```

Comment:
I really don't like the idea of passing this around as an env var (i.e. creating a global namespace for ourselves). Depending on how the env is segmented, we'll probably run into race conditions.

Is there a way to pass this around as an array of key/value pairs instead?

Reply:
I also think this would need to be `os.Setenv("$" + q.ApplicationSource.Ref, appPath)` in order to actually work. The code that does `os.Getenv` assumes a leading `$`.

Reply:
The code for os.Getenv and os.Setenv has already been updated. 

Also, while writing the proposal, I had a discussion with @alexmt offline and we agreed on using the env variables for storing `ref` variables. That way it would be easier to read the value files.